server/login.go

```diff
@@ -44,14 +44,15 @@ func GetLogin(c *gin.Context) {
 		getLoginOauth2(c)
 	}
 
+	login := ToUser(c)
+
 	// exit if authorization fails
-	// TODO(bradrydzewski) return an error message instead
 	if c.Writer.Status() != 200 {
+		log.Errorf("could not get authorization for %s.", login.Login)
+		c.Redirect(303, "/login#error=access_denied")
 		return
 	}
 
-	login := ToUser(c)
```

Comment:
Ok. Do we want to show the user whose authorization was denied. if so, we will need to get another statement inside the block, like: https://gist.github.com/oliveiradan/292460dac1dca1dcef60